File: integrations/src/main/java/services/voice/repository/CallLogRepository.java
package services.voice.repository;

import org.javers.spring.annotation.JaversSpringDataAuditable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;
import services.voice.entity.CallLog;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Repository
@JaversSpringDataAuditable
public interface CallLogRepository extends JpaRepository<CallLog, UUID> {

    Optional<CallLog> findByCallSid(String callSid);

    @Query("SELECT c FROM CallLog c WHERE c.entityName = :entityName AND c.entityId = :entityId ORDER BY c.createdAt DESC")
    List<CallLog> findByEntity(@Param("entityName") String entityName, @Param("entityId") Long entityId);

    @Query("SELECT c FROM CallLog c WHERE c.fromNumber = :fromNumber OR c.toNumber = :toNumber ORDER BY c.createdAt DESC")
    List<CallLog> findByPhoneNumber(@Param("fromNumber") String fromNumber, @Param("toNumber") String toNumber);

    @Query("SELECT c FROM CallLog c WHERE c.status = :status ORDER BY c.createdAt DESC")
    List<CallLog> findByStatus(@Param("status") String status);

    @Query("SELECT c FROM CallLog c WHERE c.direction = :direction ORDER BY c.createdAt DESC")
    List<CallLog> findByDirection(@Param("direction") String direction);
}

File: integrations/src/main/kotlin/services/voice/repository/CallLogRepository.kt
package services.voice.repository

import org.javers.spring.annotation.JaversSpringDataAuditable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository
import services.voice.entity.CallLog
import java.util.UUID

@Repository
@JaversSpringDataAuditable
interface CallLogRepository : JpaRepository<CallLog, UUID> {

    fun findByCallSid(callSid: String): CallLog?

    @Query(
        "SELECT c FROM CallLog c WHERE c.entityName = :entityName AND c.entityId = :entityId ORDER BY c.createdAt DESC"
    )
    fun findByEntity(@Param("entityName") entityName: String, @Param("entityId") entityId: Long): List<CallLog>

    @Query(
        "SELECT c FROM CallLog c WHERE c.fromNumber = :fromNumber OR c.toNumber = :toNumber ORDER BY c.createdAt DESC"
    )
    fun findByPhoneNumber(@Param("fromNumber") fromNumber: String, @Param("toNumber") toNumber: String): List<CallLog>

    @Query("SELECT c FROM CallLog c WHERE c.status = :status ORDER BY c.createdAt DESC")
    fun findByStatus(@Param("status") status: String): List<CallLog>

    @Query("SELECT c FROM CallLog c WHERE c.direction = :direction ORDER BY c.createdAt DESC")
    fun findByDirection(@Param("direction") direction: String): List<CallLog>
}
